Python SDK topic autoscaling docs (#14583)
ydb-platform/ydb: ydb/docs/en/core/reference/ydb-sdk/topic.md

  ```

{% endlist %}

### Topic autoscaling {#autoscaling}

{% list tabs group=lang %}

- Python

>   Autoscaling of a topic is could be enabled during its creation using the `auto_partitioning_settings` field:

Reviewer: ```suggestion
  Autoscaling of a topic can be enabled during its creation using the `auto_partitioning_settings` argument of `create_topic`:
```

  ```python
      driver.topic_client.create_topic(
          topic,
          consumers=[consumer],
          min_active_partitions=10,
          max_active_partitions=100,
          auto_partitioning_settings=ydb.TopicAutoPartitioningSettings(
              strategy=ydb.TopicAutoPartitioningStrategy.SCALE_UP,
              up_utilization_percent=80,
              down_utilization_percent=20,
              stabilization_window=datetime.timedelta(seconds=300),
          ),
      )
  ```

>   Changes to an existing topic are could be made using the `alter_auto_partitioning_settings` field:

Reviewer: ```suggestion
  Changes to an existing topic can be made using the `alter_auto_partitioning_settings` argument of `alter_topic`:
```

  ```python
      driver.topic_client.alter_topic(
          topic_path,
          alter_auto_partitioning_settings=ydb.TopicAlterAutoPartitioningSettings(
              set_strategy=ydb.TopicAutoPartitioningStrategy.SCALE_UP,
              set_up_utilization_percent=80,
              set_down_utilization_percent=20,
              set_stabilization_window=datetime.timedelta(seconds=300),
          ),
      )
  ```

>   The SDK supports two topic reading modes with autoscaling enabled: full support mode and compatibility mode. The reading mode could be set in the `auto_partitioning_support` field while creating the reader. Full support mode is used by default.

Reviewer: ```suggestion
  The SDK supports two topic reading modes with autoscaling enabled: full support mode and compatibility mode. The reading mode can be set in the `auto_partitioning_support` argument when creating the reader. Full support mode is used by default.
```

  ```python
  reader = driver.topic_client.reader(
      topic,
      consumer,
      auto_partitioning_support=True, # Full support is enabled
  )

  # or

  reader = driver.topic_client.reader(
      topic,
      consumer,
      auto_partitioning_support=False, # Compatibility mode is enabled
  )
  ```

>   From a practical point of view, the modes do not differ for the end user.
>   The full support mode differs from the compatibility mode in terms of who will guarantee the order of reading - the client or the server. Compatibility mode is achieved by server-side processing and, as a rule, runs slower.

Reviewer: ```suggestion
  From a practical perspective, these modes do not differ for the end user. However, the full support mode differs from the compatibility mode in terms of who guarantees the order of reading—the client or the server. Compatibility mode is achieved through server-side processing and generally operates slower.
```
